File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res=0;
        int n = s.length();
        unordered_map<char,int> freq;
        int l=0;
        int r=0;
        while(l<n){
            freq[s[l]]++;
            while(r<=l && freq[s[l]]>1){
                freq[s[r]]--;
                r++;
            }
            res=max(res,l-r+1);
            l++;
        }
        return res;
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.py
import heapq
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        pq = []
        for i in intervals:
            heapq.heappush(pq,(i.start, 1))
            heapq.heappush(pq,(i.end, 0))
        c = 0
        res = 0
        while len(pq)>0:
            po, t = heapq.heappop(pq)
            if t==1:
                c+=1
                res = max(res, c)
            else:
                c-=1
        return res